src/modules/accounting/bunker.py

```diff
@@ -0,0 +1,540 @@
+    def is_bunker_mode(
+        self,
+        blockstamp: BlockStamp,
+        frame_config: FrameConfig,
+        chain_config: ChainConfig,
+        simulated_rebase: LidoReportRebase,
+        last_finalized_slot_number: SlotNumber,
+    ) -> bool:
+        self.f_conf = frame_config
+        self.c_conf = chain_config
+        self.simulated_rebase = simulated_rebase
+        self.last_finalized_slot_number = last_finalized_slot_number
+
+        self._get_config(blockstamp)
+        self.last_report_ref_slot = self.w3.lido_contracts.accounting_oracle.functions.getLastProcessingRefSlot().call(
+            block_identifier=blockstamp.block_hash
+        )
+        if self.last_report_ref_slot == 0:
+            logger.info({"msg": "No one report yet. Bunker status will not be checked"})
+            return False
+
+        self.all_validators, self.lido_keys, self.lido_validators = self._get_lido_validators_with_others(blockstamp)
+        logger.info({"msg": f"Validators - all: {len(self.all_validators)} lido: {len(self.lido_validators)}"})
+
+        logger.info({"msg": "Checking bunker mode"})
+        frame_cl_rebase = self._get_cl_rebase_for_frame(blockstamp)
+        if frame_cl_rebase < 0:
+            logger.info({"msg": "Bunker ON. CL rebase is negative"})
+            return True
+        if self._is_high_midterm_slashing_penalty(blockstamp, frame_cl_rebase):
+            logger.info({"msg": "Bunker ON. High midterm slashing penalty"})
+            return True
+        if self._is_abnormal_cl_rebase(blockstamp, frame_cl_rebase):
+            logger.info({"msg": "Bunker ON. Abnormal CL rebase"})
+            return True
+
+        return False
+
+    def _get_config(self, blockstamp: BlockStamp):
+        """
+        Get config values from OracleDaemonConfig contract
+        """
+        config = self.w3.lido_contracts.oracle_daemon_config
+        self.b_conf = BunkerConfig(
+            Web3.to_int(
+                config.functions.get('NORMALIZED_CL_PER_EPOCH').call(block_identifier=blockstamp.block_hash)
+            ),
+            Web3.to_int(
+                config.functions.get('NORMALIZED_CL_MISTAKE_BP').call(block_identifier=blockstamp.block_hash)
+            ) / 10000,
+            Web3.to_int(
+                config.functions.get('REBASE_CHECK_NEAREST_EPOCH_DISTANCE').call(block_identifier=blockstamp.block_hash)
+            ),
+            Web3.to_int(
+                config.functions.get('REBASE_CHECK_FAR_EPOCH_DISTANCE').call(block_identifier=blockstamp.block_hash)
+            )
+        )
+
+    def _get_cl_rebase_for_frame(self, blockstamp: BlockStamp) -> Gwei:
+        logger.info({"msg": "Getting CL rebase for frame"})
+        before_report_total_pooled_ether = self._get_total_supply(blockstamp)
+
+        # Can't use from_wei - because rebase can be negative
+        frame_cl_rebase = (self.simulated_rebase.post_total_pooled_ether - before_report_total_pooled_ether) // GWEI_TO_WEI
```

Comment:
This call would include EL rewards. This is not intended by the design of the bunker mode.
Therefore, we need to make another simulated call passing zero EL vault's balance.